GoogleSignIn/Sources/GIDAppCheck/Implementations/GIDAppCheck.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <TargetConditionals.h>
+
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+
+#import <Foundation/Foundation.h>
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDSignIn.h"
+#import "GoogleSignIn/Sources/GIDAppCheck/API/GIDAppCheckProvider.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class FIRAppCheckToken;
+
+/// A list of potential error codes returned from the Google Sign-In SDK during App Check.
+typedef NS_ERROR_ENUM(kGIDSignInErrorDomain, GIDAppCheckErrorCode) {
```

Comment:
I agree -- makes sense to have a separate error domain for `GIDAppCheck`. Also, good question about whether or not the app check errors will be exposed publicly. They will, since they can be passed thru via the completion called in `-[GIDSignIn configureWithCompletion:]`. I've added the error in a new file (`GIDAppCheckError.h`) that is public.

Reply:
Let's discuss offline whether it makes sense to expose a new error domain publicly. Alternatively, we can add new codes to the existing `kGIDSignInErrorDomain`.